### PR TITLE
gh-1 Adjust client to weaviate v1

### PR DESCRIPTION
**Pull request by antas-marcin**

### CODE REVIEW

File: test/batch/batching_test.go [hunk 1]
```diff
@@ -45,85 +46,85 @@ func TestBatch_integration(t *testing.T) {
...
 		refTtoT := client.Batch().ReferencePayloadBuilder().WithFromClassName("Pizza").WithFromRefProp("otherFoods").WithFromID("97fa5147-bdad-4d74-9a81-f8babc811b09").WithToID("97fa5147-bdad-4d74-9a81-f8babc811b09").Payload()
 
 		refAtoT := &models.BatchReference{
-			From: "weaviate://localhost/actions/Soup/07473b34-0ab2-4120-882d-303d9e13f7af/otherFoods",
-			To:   "weaviate://localhost/things/97fa5147-bdad-4d74-9a81-f8babc811b09",
+			From: "weaviate://localhost/Soup/07473b34-0ab2-4120-882d-303d9e13f7af/otherFoods",
+			To:   "weaviate://localhost/97fa5147-bdad-4d74-9a81-f8babc811b09",
 		}
-		refAtoA := client.Batch().ReferencePayloadBuilder().WithFromKind(semantics.Actions).WithFromClassName("Soup").WithFromRefProp("otherFoods").WithFromID("07473b34-0ab2-4120-882d-303d9e13f7af").WithToKind(semantics.Actions).WithToID("07473b34-0ab2-4120-882d-303d9e13f7af").Payload()
+		refAtoA := client.Batch().ReferencePayloadBuilder().WithFromKind(semantics.Objects).WithFromClassName("Soup").WithFromRefProp("otherFoods").WithFromID("07473b34-0ab2-4120-882d-303d9e13f7af").WithToKind(semantics.Objects).WithToID("07473b34-0ab2-4120-882d-303d9e13f7af").Payload()
```
Comment (etiennedi) [inline test/batch/batching_test.go:127]: `.WithFromKind()` and `.WithToKind()` should be removed from ReferencePayloadBuilder

File: test/batch/batching_test.go [hunk 2]
```diff
@@ -45,85 +46,85 @@ func TestBatch_integration(t *testing.T) {
...
 		assert.Nil(t, batchErrT)
 		assert.NotNil(t, batchResultT)
 		assert.Equal(t, 2, len(batchResultT))
-		batchResultA, batchErrA := client.Batch().ActionsBatcher().WithObject(classA1).WithObject(classA2).Do(context.Background())
+		batchResultA, batchErrA := client.Batch().ObjectsBatcher().WithObject(classA1).WithObject(classA2).Do(context.Background())
 		assert.Nil(t, batchErrA)
 		assert.NotNil(t, batchResultA)
 		assert.Equal(t, 2, len(batchResultA))
 
 		time.Sleep(2.0 * time.Second) // Give weaviate time to update its index
```
Comment (etiennedi) [inline test/batch/batching_test.go:69]: unnecessary sleep

File: test/batch/batching_test.go [hunk 3]
```diff
@@ -45,85 +46,85 @@ func TestBatch_integration(t *testing.T) {
...
+			Properties: map[string]string{
 				"name":        "Beautiful",
 				"description": "Putting the game of letter soups to a whole new level.",
 			},
 		}
-		batchResultA, batchErrA := client.Batch().ActionsBatcher().WithObject(classA).Do(context.Background())
+		batchResultA, batchErrA := client.Batch().ObjectsBatcher().WithObject(classA).Do(context.Background())
 		assert.Nil(t, batchErrA)
 		assert.NotNil(t, batchResultA)
 		time.Sleep(2.0 * time.Second)
```
Comment (etiennedi) [inline test/batch/batching_test.go:114]: unnecessary sleep

File: test/batch/batching_test.go [hunk 4]
```diff
@@ -132,23 +133,23 @@ func TestBatch_integration(t *testing.T) {
...
```
Comment (etiennedi) [inline test/batch/batching_test.go:133]: unnecessary sleep

File: test/data/reference_test.go [hunk 5]
```diff
@@ -89,20 +90,20 @@ func TestData_reference_integration(t *testing.T) {
 		}
 		_, errCreateT := client.Data().Creator().WithClassName("Pizza").WithID("abefd256-8574-442b-9293-9205193737ee").WithSchema(propertySchemaT).Do(context.Background())
 		assert.Nil(t, errCreateT)
-		_, errCreateA := client.Data().Creator().WithClassName("Soup").WithID("565da3b6-60b3-40e5-ba21-e6bfe5dbba91").WithSchema(propertySchemaA).WithKind(semantics.Actions).Do(context.Background())
+		_, errCreateA := client.Data().Creator().WithClassName("Soup").WithID("565da3b6-60b3-40e5-ba21-e6bfe5dbba91").WithSchema(propertySchemaA).WithKind(semantics.Objects).Do(context.Background())
```
Comment (etiennedi) [inline test/data/reference_test.go:93]: Remove `WithKind()` from `Creator`

File: test/schema/schema_test.go [hunk 6]
```diff
@@ -21,47 +22,50 @@ func TestSchema_integration(t *testing.T) {
...
-			Class:       "Run",
-			Description: "Running from the fuzz",
+			Class:           "Run",
+			Description:     "Running from the fuzz",
+			VectorIndexType: "hnsw",
+			Vectorizer:      "text2vec-contextionary",
 		}
 
-		err := client.Schema().ClassCreator().WithClass(schemaClass).WithKind(semantics.Actions).Do(context.Background())
+		err := client.Schema().ClassCreator().WithClass(schemaClass).WithKind(semantics.Objects).Do(context.Background())
```
Comment (etiennedi) [inline test/schema/schema_test.go:62]: Remove `WithKind()` from `ClassCreator`

File: weaviate/schema/allDeleter.go [hunk 7]
```diff
@@ -19,14 +20,8 @@ func (ad *AllDeleter) Do(ctx context.Context) error {
...
-	for _, class := range schema.Actions.Classes {
-		delErr := ad.schemaAPI.ClassDeleter().WithClassName(class.Class).WithKind(semantics.Actions).Do(ctx)
-		if delErr != nil {
-			return delErr
-		}
-	}
-	for _, class := range schema.Things.Classes {
-		delErr := ad.schemaAPI.ClassDeleter().WithClassName(class.Class).WithKind(semantics.Things).Do(ctx)
+	for _, class := range schema.Classes {
+		delErr := ad.schemaAPI.ClassDeleter().WithClassName(class.Class).WithKind(semantics.Objects).Do(ctx)
```
Comment (etiennedi) [inline weaviate/schema/allDeleter.go:24]: Remove `WithKind()` from `ClassDeleter`

**Review by etiennedi — COMMENTED** [review]
Noted two things, one is not imporant and can be fixed later, but the other one should be addressed:

- There are a lot of  unnecessary sleeps in the tests (I added comments to some, but not all). This has no user impact, so this can be fixed later
- Some user-facing methods still have a `WithKind()` method on their builders. This needs to be removed. We can keep something like `kind.Object` internally for now if that makes it easier, but all user-facing mentions of kind should be removed.

File: test/batch/batching_test.go (hunk 1 above)
Comment (antas-marcin) [reply test/batch/batching_test.go:127]: done

File: test/data/reference_test.go (hunk 5 above)
Comment (antas-marcin) [reply test/data/reference_test.go:93]: done

File: weaviate/schema/allDeleter.go (hunk 7 above)
Comment (antas-marcin) [reply weaviate/schema/allDeleter.go:24]: done

File: test/schema/schema_test.go (hunk 6 above)
Comment (antas-marcin) [reply test/schema/schema_test.go:62]: done

**Review by etiennedi — APPROVED** [review]

**Pull request merged**